src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs

```diff
@@ -187,18 +191,39 @@ private void OnSurfacePainted(object? sender, SKPaintSurfaceEventArgs e)
 		paint.IsAntialias = true;
 
 		float cornerRadius = (float)_cornerRadius.BottomRight * pixelRatio;
-
+		SKPoint[] cornerRadiusPoints = Array.Empty<SKPoint>();
+
+		if (!IsCornerRadiusProportional())
+		{ 
+			cornerRadiusPoints = new SKPoint[]
+						{
+							new SKPoint((float)_cornerRadius.TopLeft * pixelRatio, (float)_cornerRadius.TopLeft * pixelRatio),
+							new SKPoint((float)_cornerRadius.TopRight * pixelRatio, (float)_cornerRadius.TopRight * pixelRatio),
+							new SKPoint((float)_cornerRadius.BottomRight * pixelRatio, (float)_cornerRadius.BottomRight * pixelRatio),
+							new SKPoint((float)_cornerRadius.BottomLeft * pixelRatio, (float)_cornerRadius.BottomLeft * pixelRatio)
+						};
+		}
 		foreach (var shadow in shadows.Where(s => !s.IsInner))
 		{
-			var skShadow = SKShadow.From(shadow, childWidth, childHeight, cornerRadius, pixelRatio);
+			var skShadow = SKShadow.From(shadow, childWidth, childHeight, cornerRadius, pixelRatio, cornerRadiusPoints);
```

Comment:
why are we still keeping the old `cornerRadius` if we have the `cornerRadiusPoints`?

Reply:
I kept cornerRadius (SkiaApi.sk_rrect_set_rect_xy) because it costs less than cornerRadiusPoints (sk_rrect_set_rect_radii).
Where the cornerRadiusPoints is only filled in if the radios are not symmetrical.